driver-sync/src/test/functional/com/mongodb/client/unified/UnifiedCrudHelper.java

```diff
@@ -692,43 +704,49 @@ public OperationResult executeCreateIndex(final BsonDocument operation) {
         String entityName = operation.getString("object").getValue();
+        BsonDocument arguments = operation.getDocument("arguments", new BsonDocument());
+        List<BsonDocument> pipeline = arguments.getArray("pipeline").stream().map(BsonValue::asDocument).collect(toList());
         ChangeStreamIterable<BsonDocument> iterable;
         if (entities.hasCollection(entityName)) {
-            iterable = entities.getCollection(entityName).watch();
+            iterable = entities.getCollection(entityName).watch(pipeline);
         } else if (entities.hasDatabase(entityName)) {
-            iterable = entities.getDatabase(entityName).watch(BsonDocument.class);
+            iterable = entities.getDatabase(entityName).watch(pipeline, BsonDocument.class);
         } else if (entities.hasClient(entityName)) {
-            iterable = entities.getClient(entityName).watch(BsonDocument.class);
+            iterable = entities.getClient(entityName).watch(pipeline, BsonDocument.class);
         } else {
             throw new UnsupportedOperationException("No entity found for id: " + entityName);
         }
 
-        for (Map.Entry<String, BsonValue> cur : operation.getDocument("arguments", new BsonDocument()).entrySet()) {
-            //noinspection SwitchStatementWithTooFewBranches
+        for (Map.Entry<String, BsonValue> cur : arguments.entrySet()) {
             switch (cur.getKey()) {
                 case "batchSize":
                     iterable.batchSize(cur.getValue().asNumber().intValue());
                     break;
+                case "pipeline":
+                    break;
                 default:
                     throw new UnsupportedOperationException("Unsupported argument: " + cur.getKey());
             }
         }
 
         return resultOf(() -> {
-            MongoCursor<BsonDocument> cursor = iterable.withDocumentClass(BsonDocument.class).cursor();
-            entities.addChangeStream(operation.getString("saveResultAsEntity").getValue(), cursor);
+            entities.addChangeStream(operation.getString("saveResultAsEntity").getValue(), iterable.cursor());
             return null;
         });
     }
 
     public OperationResult executeIterateUntilDocumentOrError(final BsonDocument operation) {
-        MongoCursor<BsonDocument> cursor = entities.getChangeStream(operation.getString("object").getValue());
+        MongoCursor<ChangeStreamDocument<BsonDocument>> cursor = entities.getChangeStream(operation.getString("object").getValue());
 
         if (operation.containsKey("arguments")) {
             throw new UnsupportedOperationException("Unexpected arguments");
         }
 
-        return resultOf(cursor::next);
+        return resultOf(() -> {
+            BsonDocumentWriter bsonDocumentWriter = new BsonDocumentWriter(new BsonDocument());
+            changeStreamDocumentCodec.encode(bsonDocumentWriter, cursor.next(), EncoderContext.DEFAULT_CONTEXT);
```

Comment:
Comment: If this is the only public use of DEFAULT_CONTEXT I probably wouldn't have made it public but I'm not against it being so either.

Reply:
I am opposed to adding it in the context of this commit, as I don't think we should mix changes to the public API with changes to a test runner.  If you feel strongly that DEFAULT_CONTEXT should be public, then let's open a separate ticket for it.

Reply:
Done.